Sources/StreamChat/Workers/ChannelUpdater.swift

```diff
@@ -46,6 +47,7 @@ class ChannelUpdater: Worker {
         channelQuery: ChannelQuery,
         isInRecoveryMode: Bool,
         onChannelCreated: ((ChannelId) -> Void)? = nil,
+        actions: ChannelUpdateActions? = nil,
```

Comment:
Resetting members and watchers needs to happen within the same save closure for avoiding multiple DB observer changes (aka better performance). The cost is having more logic in the main channel update method.